NickvisionTagger.Shared/Models/LyricService.cs

```diff
@@ -0,0 +1,133 @@
+using ATL;
+using HtmlAgilityPack;
+using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace NickvisionTagger.Shared.Models;
+
+public enum LyricProviders
+{
+    Music163 = 0, //sync lyric provider
+    Letras, //unsync lyric provider
+}
+
+/// <summary>
+/// A service for looking up and downloading lyrics for a music file
+/// </summary>
+/// <remarks>Service based off that of GiveMeLyrics: https://github.com/muriloventuroso/givemelyrics/blob/master/src/Services/LyricsFetcher.vala</remarks>
+public static class LyricService
+{
+    private static readonly HttpClient _http;
+
+    /// <summary>
+    /// Constructs a static LyricService
+    /// </summary>
+    static LyricService()
+    {
+        _http = new HttpClient();
+        _http.DefaultRequestHeaders.UserAgent.ParseAdd("Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/116.0");
+    }
+
+    /// <summary>
+    /// Gets a LyricsInfo object for a song
+    /// </summary>
+    /// <param name="title">The title of the song</param>
+    /// <param name="artist">The artist of the song</param>
+    /// <returns>The LyricInfo object if successful, else null</returns>
+    public static async Task<LyricsInfo?> GetAsync(string title, string artist)
+    {
+        if (string.IsNullOrEmpty(title) || string.IsNullOrEmpty(artist))
+        {
+            return null;
+        }
+
+        title = title.ToLower().Replace("ê", "e").Replace("á", "á").Replace("à", "à").Replace("ã", "a").Replace("ó", "o")
+            .Replace("ç", "c").Replace("í", "i").Replace("ú", "u").Replace("å", "a").Replace("ö", "o");
+        artist = artist.ToLower().Replace("ê", "e").Replace("á", "á").Replace("à", "à").Replace("ã", "a").Replace("ó", "o")
+            .Replace("ç", "c").Replace("í", "i").Replace("ú", "u").Replace("å", "a").Replace("ö", "o");
```

Comment:
I have doubts this covers all non-ASCII letters 🤔

Reply:
This is how GiveMeLyrics handled it..just replacing accented characters just because the lyrics providers change them to the non-accented version. But other non ASCII character, such as russian letters shouldn't have a problem...

Reply:
I didn't mean Russian but something like ẽ (e with tilde) or other composite characters that may appear in various languages.

Reply:
Ah ok i see...let me see if I can find a C# standard function to remove accents.

Reply:
ok fixed in latest commit